In [2]:

import pandas as pd

In [3]:
# Lets import the data

SAT = pd.read_excel(r"C:\Users\Laxman\OneDrive\Desktop\Data Files_1\Z - test - One sample - SAT Score.xlsx")

SAT.head()

Student  SAT Score
0        1        434
1        2        694
2        3        457
3        4        534
4        5        720

In [4]:
# Lets check if the data is normally distributed

SAT['SAT Score'].skew()

0.14603777698529208

Data have more than 30 data points, The SD of population is know and data is also normally distributed, Thus we can use Z-Test

For Standard Normal Test (Z - test) 
a) Large sample test (n > 30)
b) Population SD is known
c) Normally distributed
d) Max only two sample can be compared
e) compare two groups/categories of numeric variables with large sample size


In [5]:
# Lets import the z test function

from statsmodels.stats import weightstats

# Syntax - weightstats.ztest(X1 = First Sample, X2 = Second Sample)

test = weightstats.ztest(SAT['SAT Score'], x2 = None)

test

(33.10729583418216, 2.33368439508341e-240)

The p-value for the test 2.33*e^-240 ~ 0.00

Thus p-value for test < 0.05, thus, I will reject my null statement and I will go with my alternative statement i.e There is significant impact of SAT classes on average marks